### DISCUSSION TITLE
single file database for all resume | torrents ?

### DISCUSSION THREAD

**Discussion by Owyn** [transmission/transmission#6160] [Ideas]
Since 4.0 beta there was a huge work done to make loading thousands of torrents from the disk faster and that was a great optimization of functions but... how about an optimization of data?

**Tixati** stores the torrent database in a single file
**qBittorrent** stores the resume and torrent in a single sqlite database file when you enable the option for that

This would make Transmission start many times faster when there's a big number of torrents 
This would tremendously reduce the number of writes (and reads) onto the SSD which many ppl worry about, or even reduce it to just 1 write to the SSD when the program closes - an "in-memory SQLite database"can do that

**Comment by tearfur** [comment]
I like the idea 👍🏻

**Comment by Pentaphon** [comment]
Honestly, there's a lot of things that Transmission has to catch up on including data optimization and I think data optimization will come last the same way it is coming along with qBittorrent, where the SQLite DB is still a work in progress.

**Comment by jibwf** [comment]
Like this idea.

**Comment by Hunter16g** [comment]
Sounds like a good idea, however there should be a way to export some or all torrent files then.

**Comment by wally-crunk** [comment]
A database approach would help modernize how transmission "thinks" about downloaded-files-on-disk.

Suppose you have two (2) torrents that refer to the same files on disk (cross-seeding at two different trackers).
– If you move one of them, the other one breaks.
– If you verify them both, the action is repeated twice.

Today, we just have to live with these types of issues.

A database approach would be a modern way to solve both of these issues and be more efficient.
